lib/parser/models.ex

```diff
@@ -0,0 +1,385 @@
+defmodule Thrift.Parser do
+
+  defmodule Conversions do
+
+    def atomify(a) when is_atom(a) do
+      a
+    end
+
+    def atomify(l) when is_list(l) do
+      List.to_atom(l)
+    end
+
+    def atomify(s) when is_bitstring(s) do
+      String.to_atom(s)
+    end
+
+    def stringify(l) when is_list(l) do
+      List.to_string(l)
+    end
+
+    def cast(:int, val) do
+      val
+    end
+
+    def cast(:double, val) do
+      val
+    end
+
+    def cast(:string, val) do
+      List.to_string(val)
+    end
+
+    def cast({:set, type}, val) do
+      MapSet.new(val, &cast(type, &1))
+    end
+
+    def cast({:map, {key_type, val_type}}, val) do
+      Enum.into(val, %{}, fn {k, v} ->
+        {cast(key_type, k), cast(val_type, v)}
+      end)
+    end
+
+    def cast(_, val) do
+      val
+    end
+  end
+
+
+  defmodule Models do
+
+    defmodule Namespace do
+      @moduledoc """
+      A thrift namespace.
+      The namespace is a language-specific place where the generated structs are
+      placed.
+      """
+      defstruct name: nil, path: nil
+      import Thrift.Parser.Conversions
+
+      def new(name, path) do
+        %Namespace{name: atomify(name), path: stringify(path)}
```

Comment:
I'm pretty sure we atomify things that are already atoms (like nil), but I can safely remove the bitstring versions. 

Don't the guard clauses tell the story here? What would a spec add?